awot/dropsondes/dropsondes.py

```diff
@@ -8,651 +8,698 @@
+        sounding_data, skip_header=8, usecols=range(0, 8),
+        dtype=float, delimiter=None, autostrip=True,
+        missing_values='-9999.00', unpack=True, usemask=True)
+
+    u = -ws*np.sin(np.radians(wd))
+    v = -ws*np.cos(np.radians(wd))
+    # RH = tC._dewpoint_to_RH(T+273.15, Td+273.15)
+
+    mask = T.mask
+    T = T[~mask]
+    TD = Td[~mask]
+    P = p[~mask]
+    H = h[~mask]
+    RH = RH[~mask]
+
+    mask = u.mask
+    U = u[~mask]
+    V = v[~mask]
+
+    data = dict()
+    data['Header'] = header
+    data['Temperature'] = T
+    data['Dewpoint'] = Td
+    data['Pressure'] = p
+    data['Relative Humidity'] = RH
+    data['U Component'] = U
+    data['V Component'] = V
+    data['Height'] = h
+    data['Type'] = 'radioSonde'
+
+    fp.close()
+
+    # type = 'radioSonde'
+
+    return data
+
+def get_dropsonde_data(filePath):
 
-        p, T, Td, RH, u, v, h = np.genfromtxt(
-            sounding_data, skip_header=15, usecols=(1, 2, 3, 4, 5, 6, 14),
-            dtype=float, missing_values='9999.0', unpack=True, usemask=True)
+    '''
+    method to retrieve dropsonde data
 
-        # mask incoming T and dewpoint data
+    Inputs:
 
-        T = np.ma.masked_greater_equal(T, 999.0)
-        Td = np.ma.masked_greater_equal(Td, 999.0)
-        RH = np.ma.masked_greater_equal(RH, 999.0)
-        h = np.ma.masked_greater_equal(h, 99999.0)
+    file path
+    #============
 
-        mask = T.mask
-        T = T[~mask]
-        TD = Td[~mask]
-        P = p[~mask]
-        H = h[~mask]
-        RH = RH[~mask]
+    output:
 
-        mask = u.mask
-        U = u[~mask]
-        V = v[~mask]
+    #Dictionary with variable for use in plotting
 
-        data = dict()
-        data['Header'] = header
-        data['Temperature'] = T
-        data['Dewpoint'] = TD
-        data['Pressure'] = P
-        data['Relative Humidity'] = RH
-        data['U Component'] = U
-        data['V Component'] = V
-        data['Height'] = H
-        data['Type'] = 'dropsonde'
+    '''
 
-        return data
+    data = filePath
+    fp = open(data, 'r')
+    lines = fp.readlines()
+    header = lines[1]
+    fp.close()
+    fp = open(data, 'r')
+    sounding_data = fp
+
+    p, T, Td, RH, u, v, h = np.genfromtxt(
+        sounding_data, skip_header=15, usecols=(1, 2, 3, 4, 5, 6, 14),
+        dtype=float, missing_values='9999.0', unpack=True, usemask=True)
+
+    # mask incoming T and dewpoint data
+
+    T = np.ma.masked_greater_equal(T, 999.0)
+    Td = np.ma.masked_greater_equal(Td, 999.0)
+    RH = np.ma.masked_greater_equal(RH, 999.0)
+    h = np.ma.masked_greater_equal(h, 99999.0)
+
+    mask = T.mask
+    T = T[~mask]
+    TD = Td[~mask]
+    P = p[~mask]
+    H = h[~mask]
+    RH = RH[~mask]
+
+    mask = u.mask
+    U = u[~mask]
+    V = v[~mask]
+
+    data = dict()
+    data['Header'] = header
+    data['Temperature'] = T
+    data['Dewpoint'] = TD
+    data['Pressure'] = P
+    data['Relative Humidity'] = RH
+    data['U Component'] = U
```

Comment:
This is a personal choice, but I don't like using spaces in the name, e.g. 'U Component' to 'u_component'. I also tend to use all lower case for the keys.